includes/purge-set-object.batch.inc

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * @file
+ * Batch functions for purging ingested objects via set.
+ */
+
+
+/**
+ * Batch to purge objects ingested via a batch set.
+ *
+ * @param int $set_id
+ *   Batch set ID.
+ */
+function islandora_batch_batch_purge_object_in_set($set_id) {
+  module_load_include('inc', 'islandora_batch', 'includes/db');
+  $batch = array(
+    'title' => t('Purge objects from set'),
+    'finished' => 'islandora_batch_batch_purge_object_in_set_finished',
+    'init_message' => t('Initializing...'),
+    'operations' => array(),
+    'progress_message' => t('Time elapsed: @elapsed.'),
+    'file' => drupal_get_path('module', 'islandora_batch') . '/includes/purge-set-object.batch.inc',
+  );
+
+  $batch['operations'][] = array('islandora_batch_batch_purge_object_in_set_operation', array($set_id));
+  batch_set($batch);
+}
+
+/**
+ * Batch operation to purge object ingested via batch.
+ *
+ * @param int $set_id
+ *   Batch Set ID.
+ *
+ * @param array $context
+ *   Batch context.
+ */
+function islandora_batch_batch_purge_object_in_set_operation($set_id, &$context) {
+  module_load_include('inc', 'islandora_batch', 'includes/ingest.batch');
+  module_load_include('inc', 'islandora_batch', 'includes/db');
+  $sandbox =& $context['sandbox'];
+  if (!islandora_batch_get_lock($set_id)) {
+    // Failed to acquire the lock... Try again in another iteration...
+    $context['finished'] = 0;
+    return;
+  }
+
+  if (!isset($sandbox['total'])) {
+    $sandbox['items'] = islandora_batch_get_queued_items_for_set($set_id);
+    $sandbox['total'] = count($sandbox['items']);
+    $sandbox['progress'] = 0;
+    if ($sandbox['total'] === 0) {
+      $context['message'] = t('Nothing to process...');
+      return;
+    }
+  }
+  // Might be increased to avoid bootstrapping overhead.
+  $slice_size = 30;
+  $slice_stop = $sandbox['progress'] + $slice_size;
+
+  for (; $sandbox['progress'] < $slice_stop && $sandbox['progress'] < $sandbox['total']; $sandbox['progress']++) {
```

Comment:
Likely have to add a no-op expression at the start... Coder is complaining: https://travis-ci.org/Islandora/islandora_batch/jobs/156838943#L374 ... Or,  could move the `$slice_stop = $sandbox['progress'] + $slice_size` bit there?